### PR TITLE
chinese for kubectl translate

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@brendandburns maybe ```hack/update-translations.sh``` need  do in docker ?

**Comment by zjj2wry** [comment]
@shiywang hey, the translate work maybe not complete. did you have interest continue or help review？

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 1]
```diff
@@ -79,6 +79,10 @@ msgstr "取消设置 kubeconfig 文件中的一个单个值"
 msgid "Update the annotations on a resource"
 msgstr "更新一个资源的注解"
 
+#: pkg/kubectl/cmd/rename_context.go:64
+msgid "Renames a context from the kubeconfig file"
+msgstr "修改 kubeconfig 文件中的一个上下文名称"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 上下文 能否换个词，很奇怪。
或者直接参考 zh_TW中的翻译，直接就是context

**Review by zhangxiaoyu-zidif — CHANGES_REQUESTED** [review]

**Comment by shiywang** [comment]
@zjj2wry sorry, I don't have time working on this right now : )

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po (hunk 1 above)
Comment (zjj2wry) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 嗯嗯，关键字保留可能更好，WDYT? @shiywang

Comment (zjj2wry) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: @zhangxiaoyu-zidif 你有没有兴趣一起翻译？

Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: It is hard to refuse a request from a girl. :)

**Comment by zjj2wry** [comment]
@xiangpengzhao @shiywang @zhangxiaoyu-zidif hey, i will compele this recently. and i will appreciate if you can help review :)

**Comment by zhangxiaoyu-zidif** [comment]
@xiangpengzhao 
would you mind adding ok-to-test?

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 2]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\t  kubectl create configmap my-config --from-file=key1=/path/to/bar/file1.txt --from-file=key2=/path/to/bar/file2.txt\n"
+"\n"
+"\t\t  # Create a new configmap named my-config with key1=config1 and key2=config2\n"
+"\t\t  kubectl create configmap my-config --from-literal=key1=config1 --from-literal=key2=config2"
+msgstr ""
+"\n"
+"\t\t  # 通过目录 bar 创建一个名称为 my-config 的 configmap\n"
+"\t\t  kubectl create configmap my-config --from-file=path/to/bar\n"
+"\n"
+"\t\t  # 创建一个 名称为 my-config 的 configmap 并指定 keys 而不是使用磁盘上所在的文件名\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 多了一个空格

Comment (zhangxiaoyu-zidif) [reply translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 创建一个 名称为
->
创建一个名称为

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 3]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\t  kubectl create configmap my-config --from-file=path/to/bar\n"
+"\n"
+"\t\t  # Create a new configmap named my-config with specified keys instead of file basenames on disk\n"
+"\t\t  kubectl create configmap my-config --from-file=key1=/path/to/bar/file1.txt --from-file=key2=/path/to/bar/file2.txt\n"
+"\n"
+"\t\t  # Create a new configmap named my-config with key1=config1 and key2=config2\n"
+"\t\t  kubectl create configmap my-config --from-literal=key1=config1 --from-literal=key2=config2"
+msgstr ""
+"\n"
+"\t\t  # 通过目录 bar 创建一个名称为 my-config 的 configmap\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 目录
->
文件夹

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 4]
```diff
@@ -2,88 +2,2812 @@
...
+"\n"
+"\t\t# Convert the live state of the resource specified by 'pod.yaml' to the latest version\n"
+"\t\t# and print to stdout in json format.\n"
+"\t\tkubectl convert -f pod.yaml --local -o json\n"
+"\n"
+"\t\t# Convert all files under current directory to latest version and create them all.\n"
+"\t\tkubectl convert -f . | kubectl create -f -"
+msgstr ""
+"\n"
+"\t\t# 将 ’pod.yaml' 转换为最新版本并打印到 stdout.\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: ’pod.yaml'
这个是中英文标点混用？

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 5]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\tkubectl convert -f pod.yaml\n"
+"\n"
+"\t\t# 将 ‘pod.yaml' 指定的资源的实时状态转换为最新版本\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: ’pod.yaml'
这个是中英文标点混用？

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 6]
```diff
@@ -2,88 +2,2812 @@
...
+msgid ""
+"\n"
+"\t\t# Create a ClusterRole named \"pod-reader\" that allows user to perform \"get\", \"watch\" and \"list\" on pods\n"
+"\t\tkubectl create clusterrole pod-reader --verb=get,list,watch --resource=pods\n"
+"\n"
+"\t\t# Create a ClusterRole named \"pod-reader\" with ResourceName specified\n"
+"\t\tkubectl create clusterrole pod-reader --verb=get,list,watch --resource=pods --resource-name=readablepod"
+msgstr ""
+"\n"
+"\t\t# 创建一个名为 \"pod-reader\" 的 ClusterRole, 允许用户在 pods 上执行 “get\", \"watch\" 和 “list\"\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: “get\", \"watch\" 和 “list\"\n"
标点混用？

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 7]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\t# 强制删除名为 foo 的 pod\n"
+"\t\tkubectl delete pod foo --grace-period=0 --force\n"
+"\n"
+"\t\t# 删除所有 pods\n"
+"\t\tkubectl delete pods --all"
+
+#: pkg/kubectl/cmd/describe.go:54
+msgid ""
+"\n"
+"\t\t# Describe 一个 node\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 这里被误修改了？

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 8]
```diff
@@ -2,88 +2,2812 @@
...
+msgid ""
+"\n"
+"\t\t# Drain node \"foo\", even if there are pods not managed by a ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet on it.\n"
+"\t\t$ kubectl drain foo --force\n"
+"\n"
+"\t\t# As above, but abort if there are pods not managed by a ReplicationController, ReplicaSet, Job, DaemonSet 或者 StatefulSet, and use a grace period of 15 minutes.\n"
+"\t\t$ kubectl drain foo --grace-period=900"
+msgstr ""
+"\n"
+"\t\t# 清理节点 \"foo\", 即使很多 pods 没有被一个在 node 上的 ReplicationController, ReplicaSet, Job, DaemonSet 或者 StatefulSet 管理.\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 这个drain，是不是‘’驱逐‘’更合适？@tenqm

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 9]
```diff
@@ -2,88 +2,2812 @@
...
+"\n"
+"\t\t# Get output from the first pod of a ReplicaSet named nginx\n"
+"\t\tkubectl attach rs/nginx\n"
+"\t\t"
+msgstr ""
+"\n"
+"\t\t# 从运行中pod 123456-7890 获取执行 'date' 的输出, 默认使用第一个容器\n"
+"\t\tkubectl attach 123456-7890\n"
+"\n"
+"\t\t# Get output from ruby-container from pod 123456-7890\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 遗漏了?未翻译

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 10]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\t# 使用 JSON 格式化输出显示一个单独的 pod.\n"
+"\t\tkubectl get -o json pod web-pod-13je7\n"
+"\n"
+"\t\t# 显示一个被 \"pod.yaml\" 中的 type 和 name 标识的 pod 并使用 JSON 格式化输出.\n"
+"\t\tkubectl get -f pod.yaml -o json\n"
+"\n"
+"\t\t# 只返回被指定 pod 中 phase 的值.\n"
+"\t\tkubectl get -o template pod/web-pod-13je7 --template={{.status.phase}}\n"
+"\n"
+"\t\t# 显示所有的 replication controllers 和 services 并格式化 输出.\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 并格式化 输出
->
并格式化输出

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 11]
```diff
@@ -2,88 +2,2812 @@
...
+"\t\t# 显示一个或者更多 resources 通过它们的 type 和 names.\n"
+"\t\tkubectl get rc/web service/frontend pods/web-pod-13je7\n"
+"\n"
+"\t\t# 使用不同的 types 显示所有 resources.\n"
+"\t\tkubectl get all"
+
+#: pkg/kubectl/cmd/portforward.go:53
+msgid ""
+"\n"
+"\t\t# 在本地监听端口 5000 和 6000 , forwarding 数据 to/from 在 pod 5000 和 6000 端口上n\t\tkubectl port-forward mypod 5000 6000\n"
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: 这个是误翻译吗？
这是id的位置

**Comment by zjj2wry** [comment]
@zhangxiaoyu-zidif thank you review this, updated. ptal

File: translations/kubectl/zh_CN/LC_MESSAGES/k8s.po [hunk 12]
```diff
@@ -2,88 +2,2813 @@
...
+"\t\twill make the node schedulable again.\n"
+"\n"
+"\t\t![Workflow](http://kubernetes.io/images/docs/kubectl_drain.svg)"
+
+#: pkg/kubectl/cmd/edit.go:56
+msgid ""
+"\n"
+"\t\tEdit a resource from the default editor.\n"
+"\n"
+"\t\tThe edit comm
```
Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: Those lines are not translated？

Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: Do you prepare to translate them later in other PR?

Comment (zhangxiaoyu-zidif) [inline translations/kubectl/zh_CN/LC_MESSAGES/k8s.po]: current-context ?

**Comment by zhangxiaoyu-zidif** [comment]
It's really a huge PR.
From CN translation viewpoint it deserves a LGTM label.
I do not know this PR containing some untranslated lines, is that OK to merge.
But for translated part, that looks good to me.
Thanks for your contributions ! @zjj2wry

**Comment by zjj2wry** [comment]
plan create another PR translate some new command. when chinese coverage majority command, we will have more volunteer for contributor and reviewer : )

@zhangxiaoyu-zidif thank you review this, ptal
/assign @brendanburns for approve, thanks

**Review by zhangxiaoyu-zidif — APPROVED** [review]

**Pull request merged**